packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1761,13 +1761,18 @@ abstract class RenderBox extends RenderObject {
       }
       // verify that the size is within the constraints
       if (!constraints.isSatisfiedBy(_size)) {
-        throw FlutterError(
-          '$runtimeType does not meet its constraints.\n'
-          'Constraints: $constraints\n'
-          'Size: $_size\n'
-          'If you are not writing your own RenderBox subclass, then this is not '
-          'your fault. Contact support: https://github.com/flutter/flutter/issues/new?template=BUG.md'
-        );
+        throw FlutterError(<DiagnosticsNode>[
+          ErrorMessage(
+            '$runtimeType does not meet its constraints.\n'
```

Comment:
One thing I would like to figure out now before I start refactoring the 200 existing errors again is whether we can use this convention where FlutterError contains both the summary line and the rest of the error message with the exception of the hints. 
We will detect an extra line break at the end of the first paragraph to determine the boundary between the summary and rest of the description. We can add a lint to warn if the convention is not used. As the ErrorMessage body is required to be a string literal anyway to work with the kernel transformer, the lint will be 100% robust.

Reply:
I could go either way on this. @goderbauer, do you have a preference? The difference as I understand it is between these:

```dart
  throw FlutterError(<DiagnosticsNode>[
    ErrorMessage('$runtimeType does not meet its constraints'),
    ErrorDescription(
      'Constraints: $constraints\n'
      'Size: $_size'
    ),
    ErrorHint(
      'If you are not writing your own RenderBox subclass, then this is not '
      'your fault. Contact support: https://github.com/flutter/flutter/issues/new?template=BUG.md'
    ),
  ]);
```

```dart
  throw FlutterError(<DiagnosticsNode>[
    ErrorMessage(
      '$runtimeType does not meet its constraints\n'
      'Constraints: $constraints\n'
      'Size: $_size'
    ),
    ErrorHint(
      'If you are not writing your own RenderBox subclass, then this is not '
      'your fault. Contact support: https://github.com/flutter/flutter/issues/new?template=BUG.md'
    ),
  ]);
```

Reply:
That is correct although option #2 should be written as:
```dart
 throw FlutterError(<DiagnosticsNode>[
    ErrorMessage(
      '$runtimeType does not meet its constraints\n'
      '\n'
      'Constraints: $constraints\n'
      'Size: $_size'
    ),
    ErrorHint(
      'If you are not writing your own RenderBox subclass, then this is not '
      'your fault. Contact support: https://github.com/flutter/flutter/issues/new?template=BUG.md'
    ),
  ]);
```
to disambiguate the separation between the summary from the body of the error message. Good news is if someone didn't add the line break to disambiguate that could be a lint error.
This is similar to what happens for a dartdoc comment if you fail to add an extra line break after the first line. I'm fine with either option.

Reply:
What advantage do we gain by going with the blank-line convention? Looking at it, it just seems like more ugly code (`\n\n`) to write?

Looking at them side by side I find approach 1 cleaner to read. Although, it looks like a more appropriate name for the parts may be `ErrorSummary`, `ErrorDetails`, instead of `ErrorMessage`, `ErrorDescription`? Summary hopefully implying that it's supposed to be kept short?

If there's advantage in using approach 2, we absolutely do need a lint for this. That a dartdoc should start with a one-sentence summary as its own paragraph is something I have to frequently point out in code reviews - so I'd prefer it if things like these could be pointed out automatically.